File: week6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")


def text_info(txt):

    l = 0
    w = 1
    s = 0

    for t in txt:
        if t.isalpha():
            l += 1
        else:
            pass

        if t == ' ':
            w += 1
        else:
            pass

        if t in ['.', '!', '?']:
            s += 1
        else:
            pass

    return l, w, s


num_letters, num_words, num_sentences = text_info(text)


L = num_letters * 100.0 / num_words
S = num_sentences * 100.0 / num_words
index = round(0.0588 * L - 0.296 * S - 15.8)


if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int numletters(string s);
int numwords(string s);
int numsentences(string s);

int main(void)
{
    string text = get_string("Text: ");

    float L = (numletters(text) * 100.0 / numwords(text));
    float S = (numsentences(text) * 100.0 / numwords(text));
    float index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}

int numletters(string s)
{
    int l = 0;

    for (int i = 0, j = strlen(s); i < j; i++)
    {
        if ((s[i] >= 'A') && (s[i] <= 'z'))
        {
            l += 1;
        }
        else
        {
            l += 0;
        }
    }
    return l;
}

int numwords(string s)
{
    int w = 0;

    for (int i = 0, j = strlen(s); i < j; i++)
    {
        if (s[i] == ' ')
        {
            w += 1;
        }
        else
        {
            w += 0;
        }
    }
    return w + 1;
}

int numsentences(string s)
{
    int sent = 0;

    for (int i = 0, j = strlen(s); i < j; i++)
    {
        if ((s[i] == '.') || (s[i] == '!') || (s[i] == '?'))
        {
            sent += 1;
        }
        else
        {
            sent += 0;
        }
    }
    return sent;
}
